pontoon/sync/models.py

```diff
@@ -45,6 +50,36 @@ def finished(self):
     def get_absolute_url(self):
         return reverse('pontoon.sync.logs.details', kwargs={'sync_log_pk': self.pk})
 
+    def fix_stats(self):
+        """
+        Recalculate any broken stats when sync task is finished. This is a
+        temporary fix for bug 1310668.
+        """
+        if not self.finished:
+            return
+
+        # total_strings missmatch between TranslatedResource & Resource
+        for t in TranslatedResource.objects.exclude(total_strings=F('resource__total_strings')).select_related('resource'):
+            t.total_strings = t.resource.total_strings
+            t.save()
+
+        # total_strings missmatch in ProjectLocales within the same project
+        for p in Project.objects.available():
+          if ProjectLocale.objects.filter(project=p).values("total_strings").distinct().count() > 1:
+            for pl in ProjectLocale.objects.filter(project=p):
```

Comment:
On a quick look I'm not sure how. :)

Since this is only temporary (sic!) and doesn't perform slowly, I'm keeping it unchanged.